### DISCUSSION TITLE
How to define indexes for Mongodb collections in Dove (v5)

### DISCUSSION THREAD

**Discussion by nathanbrizzee-cdcr** [feathersjs/feathers#2757] [Q&A]
I was wondering what the best (Feathers) way is to define Mongodb collection indexes in the Dove release (v5)?  In v4, indexes are defined in the model file.  With the new schemas / resolvers files, should there be something in the migrations folder for Mongo?  Since Mongo creates collections on the fly, the indexes need to be applied after creation.  Is this something we should do in the constructor of the class?

**Comment by suleimanas** [comment]
Late answer:

for example in the users.class
```
import { MongoDBService } from "@feathersjs/mongodb";

export class UserService extends MongoDBService {}

export const getOptions = (app) => {
    return {
        paginate: app.get("paginate"),

        Model: app.get("mongodbClient").then((db) => {
            const model = db.collection("users");

            // your indexes
            model.createIndex({ email: 1 }, { unique: true }).catch(() => {});

            return model;
        }),
    };
};

```
